platform-core-module/src/main/java/com/softicar/platform/core/module/email/converter/EmailToPdfConverter.java

```diff
@@ -102,39 +129,98 @@ private boolean isInline(Part part) {
 		}
 	}
 
-	private void render(Part part, OutputStream output) {
+	private void render(Part part) {
 
 		try {
-			String content = (String) part.getContent();
-			if (part.isMimeType(MimeType.TEXT_HTML.getIdentifier())) {
-				renderHtml(content, output);
-			} else if (part.isMimeType(MimeType.TEXT_PLAIN.getIdentifier())) {
-				var html = "<html><body><pre>" + escape(content) + "</pre></body></html>";
-				renderHtml(html, output);
+			Object content = part.getContent();
+
+			if (part.isMimeType("text/*")) {
+				String contentString = (String) content;
+				if (part.isMimeType(MimeType.TEXT_HTML.getIdentifier())) {
+					html.append(contentString);
+				} else if (part.isMimeType(MimeType.TEXT_PLAIN.getIdentifier())) {
+					html.append("<html><body><pre>" + escapePlainTextForHtml(contentString) + "</pre></body></html>");
+				}
+			}
+
+			else if (part.isMimeType("image/*")) {
+				String base64Data = getBase64Data((InputStream) content);
+				String mimeType = determineMimeType(part);
```

Comment:
hmm, for what reason do we bother with the mime type, couldn't we just take it as it is?

Reply:
Unfortunately, we cannot:
```
data:%s;base64,%s
     ^
   that String is inserted here
```
...and we can't allow those additional attributes (e.g. `; name="filename.ext"`) from the part content type there.

Reply:
ok, got it